Code/Core/Process/SharedMemory.cpp

```diff
@@ -6,17 +6,67 @@
 #include "Core/PrecompiledHeader.h"
 
 #include "SharedMemory.h"
+#include "Core/Env/Assert.h"
+#include "Core/Strings/AString.h"
 
 #if defined( __WINDOWS__ )
     #include <windows.h>
+#elif defined(__LINUX__) || defined(__APPLE__)
+	#include <fcntl.h>
+	#include <unistd.h>
+	#include <sys/mman.h>
 #endif
 
+
+#if defined(__LINUX__) || defined(__APPLE__)
+namespace
```

Comment:
Note that SharedMemory might not be required for OSX and Linux.
It's currently only used by one feature on Windows, which is to pass info between child processes when FASTBuild is run from Visual Studio. We need two processes because of the way VS terminates compilation.